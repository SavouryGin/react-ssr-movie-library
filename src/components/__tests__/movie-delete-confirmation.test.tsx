import MovieDeleteConfirmation from 'components/movie/delete-confirmation';
import React from 'react';
import userEvent from '@testing-library/user-event';
import { render, screen } from '@testing-library/react';

const testProps = {
  className: 'test-class',
  onConfirm: jest.fn(),
};

describe('MovieDeleteConfirmation component:', () => {
  it('Should matche the snapshot', () => {
    const { asFragment } = render(<MovieDeleteConfirmation {...testProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should call the onConfirm handler', () => {
    render(<MovieDeleteConfirmation {...testProps} />);
    const button = screen.getByText('Confirm');
    userEvent.click(button);
    expect(testProps.onConfirm).toHaveBeenCalledTimes(1);
  });
});
