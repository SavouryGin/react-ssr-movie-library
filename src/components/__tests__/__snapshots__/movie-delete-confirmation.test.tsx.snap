// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDeleteConfirmation component: matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="test-class"
  >
    <p>
      Are you sure you want to delete this movie?
    </p>
    <button
      class=""
      title="Confirm movie delete"
      type="button"
    >
      <span
        class=""
      >
        Confirm
      </span>
    </button>
  </div>
</DocumentFragment>
`;
